devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -1431,6 +1521,33 @@ void YarpRobotLoggerDevice::recordVideo(const std::string& cameraName, VideoWrit
 
 void YarpRobotLoggerDevice::run()
 {
+
+    bool wait = (m_RealTimeSchedulingStrategy == RealTimeSchedulingStrategy::EarlyWakeUp
+                 || m_RealTimeSchedulingStrategy == RealTimeSchedulingStrategy::EarlyWakeUpAndFIFO)
+                    ? true
+                    : false;
+    std::chrono::nanoseconds now = BipedalLocomotion::clock().now();
+
+    while (wait)
+    {
+
+        if (now >= m_resumeTime)
+        {
+            wait = false;
+        } else
+        {
+            if ((m_resumeTime - now) > (m_awakeningTime / 2))
+            {
+                // still have time to sleep
+                BipedalLocomotion::clock().sleepFor(m_awakeningTime / 10);
+            } else
+            {
+                // just half time missing, busy wait
```

Comment:
I think you can remove the empty else

Reply:
solved by [3e9a0ce](https://github.com/ami-iit/bipedal-locomotion-framework/pull/883/commits/3e9a0ceab085d7dc156501dcc45cfda6b24f5340)